.NET/Microsoft.Recognizers.Text.DateTime/Utilities/TasksModeSetHandler.cs

```diff
@@ -0,0 +1,292 @@
+            {
+                if (refDate.DayOfWeek == DayOfWeek.Saturday)
+                {
+                    result.FutureResolution = new Dictionary<string, string>
+                    {
+                        { TimeTypeConstants.DATE, DateTimeFormatUtil.FormatDate((DateObject)refDate.AddDays(2)) },
+                    };
+
+                    result.PastResolution = new Dictionary<string, string>
+                    {
+                        { TimeTypeConstants.DATE, DateTimeFormatUtil.FormatDate((DateObject)refDate.AddDays(2)) },
+                    };
+                }
+                else if (refDate.DayOfWeek == DayOfWeek.Sunday)
+                {
+                    result.FutureResolution = new Dictionary<string, string>
+                    {
+                        { TimeTypeConstants.DATE, DateTimeFormatUtil.FormatDate((DateObject)refDate.AddDays(1)) },
+                    };
+
+                    result.PastResolution = new Dictionary<string, string>
+                    {
+                        { TimeTypeConstants.DATE, DateTimeFormatUtil.FormatDate((DateObject)refDate.AddDays(1)) },
+                    };
+                }
+                else
+                {
+                    result.FutureResolution = new Dictionary<string, string>
+                    {
+                        { TimeTypeConstants.DATE, DateTimeFormatUtil.FormatDate((DateObject)refDate) },
+                    };
+
+                    result.PastResolution = new Dictionary<string, string>
+                    {
+                        { TimeTypeConstants.DATE, DateTimeFormatUtil.FormatDate((DateObject)refDate) },
+                    };
+                }
+            }
+            else if (result.Timex.StartsWith(TasksModeConstants.GeneralPeriodPrefix))
+            {
+                result.FutureResolution = new Dictionary<string, string>
+                {
+                    { TimeTypeConstants.DATE, DateTimeFormatUtil.FormatDate((DateObject)refDate) },
+                };
+
+                result.PastResolution = new Dictionary<string, string>
+                {
+                    { TimeTypeConstants.DATE, DateTimeFormatUtil.FormatDate((DateObject)refDate) },
+                };
+            }
+            else if (result.Timex.StartsWith(TasksModeConstants.FuzzyYear))
+            {
+                var timexRes = TimexResolver.Resolve(new[] { result.Timex }, refDate);
+
+                string value = timexRes.Values[1].Value;
+
+                var resKey = TimeTypeConstants.DATETIME;
+
+                if (!result.Timex.Contains(TasksModeConstants.TimeTimexPrefix))
+                {
+                    resKey = TimeTypeConstants.DATE;
+                }
+
+                var futureValue = refDate.AddDays(7);
+
+                if (DateTimeFormatUtil.FormatDate(futureValue).Equals(value.Substring(0, 10)) && result.Timex.StartsWith(TasksModeConstants.FuzzyYearAndWeek))
```

Comment:
Why these exact indexes for substring? It would be better if they are constants defined somewhere or calculated based on some language config.

Reply:
value is language independent; it is datetime string, and specific indexing is for extracting only date part from datetime

Reply:
So we shoud define constants for them and add a comment describing their value and its calculation.

Reply:
created constants for start and end value index and added comments for the same.